### ISSUE TITLE
pretraining settings of AVT_vgg40k

### ISSUE THREAD

**Issue by yyyxcleo** [haihuangcode/CMG#13]
您好，想请问一下您提供的AVT_vgg40k_size400.pt文件的pretraining settings是什么呢？我使用您提供的pretrain代码与数据集跑了一下，并用ave_av_downstream验证，得到的video evaluation result(acc)是41%，而使用您给的checkpoint文件跑出来的video evaluation result是52%

**Comment by haihuangcode** [comment]
我在readme中有提到下面这一段"There is no validation set for the pre-training process, in this paper it is done by testing the performance of each model on the downstream of the [AVE](https://github.com/YapengTian/AVE-ECCV18), and the model with the best performance tests the rest of the downstream tasks, so the [AVE](https://github.com/YapengTian/AVE-ECCV18) can be regarded as a validation set and the model with the best pre-training appears in the first 5 epochs."
你的模型是按上面这种方式测的吗？

**Comment by yyyxcleo** [comment]
是的，我执行的./ave.sh文件

**Comment by yyyxcleo** [comment]
![image](https://github.com/user-attachments/assets/24be66bd-9ab0-4376-ba21-1cb17f1dbcc8)
结果是这样的

**Comment by haihuangcode** [comment]
你跑了哪些epoch呢？0-5的checkpoint都测了吗？

**Comment by yyyxcleo** [comment]
是的，最高的是42%，想问一下您提供的AVT_vgg40k_size400.pt文件的训练参数与pretrain.sh中是一致的吗，我多跑几轮试一下，还有一个问题想请教您，在av_downstream任务中，主要的metric就是video_evaluation_acc吗

**Comment by haihuangcode** [comment]
是一致的；ave的a2v下游metric就是video_evaluation_acc。
你多跑几轮效果不一定会变好，我印象里一般是epoch2-4（从0开始计）的模型是效果最好的，后面就有点过拟合了。
而且模型和训练的机器可能也有点关系，论文里记录的结果是用A100 80G训练得到的，而github提供的checkpoint是我之后用4090重新训的，代码是一样的，但是模型效果在不同的下游上稍微有点区别，平均下来也差不多。

**Comment by yyyxcleo** [comment]
好的我了解了，谢谢您！

**Comment by haihuangcode** [comment]
我刚才仔细看了以下pretrain.py，不知道什么时候把mi_first_forward函数的返回值顺序搞错了，我很抱歉，现在的pretrain.py已经修正，你可以更换以下这个代码再试试。不好意思🤦‍

**Comment by yyyxcleo** [comment]
哈哈好的我试一下，非常感谢！

**Comment by yyyxcleo** [comment]
您好，想请问一下cross_cpc这里softmax和lsoftmax的dim应该是0还是-1呢，我一直把dim=-1加上了，不知道是不是这里的问题，因为跑出来的模型结果还是不太好
![image](https://github.com/user-attachments/assets/fa5a5db8-008d-4a79-9ea8-01ef05e784b4)

**Comment by haihuangcode** [comment]
softmax和lsoftmax的dim默认是-1，你加上去应该不影响结果。
cross_cpc里的softmax只影响统计的acc，不影响模型的loss以及最终的训练得到的模型，不过这里确实有个统计错误，argmax的dim应该为1。
![image](https://github.com/user-attachments/assets/e731bb42-6d5c-415a-9eb1-d76d1f769314)
而lsoftmax默认-1是没问题的。
要不你给这个邮箱(haihuangcode@outlook.com)发下你微信号？我们沟通方便点。

**Comment by wy-coliney** [comment]
请问这个问题你解决了吗？我也遇到了相同的模型performance问题。大概只有41%出头 @yyyxcleo

**Comment by haihuangcode** [comment]
> 请问这个问题你解决了吗？我也遇到了相同的模型performance问题。大概只有41%出头 @yyyxcleo

你好，我这2天重新训练模型进行了测试，并且更新了一版新的pretrain.py，主要是将编码次数减少了，能加速预训练接近1倍的速度。
然后我对之前的pretrain.py和新的pretrain.py预训练的模型都进行了测试，大致如下（因为是按epoch保存的模型，所以不一定是最好的效果，不代表新的效果就更好。如果按step数，比如每隔100/200个step保存一个模型应该能得到更好的结果，我这里简单测试了一下ave的va和av）
**之前的预训练代码训练得到的模型，后面的数字代表epoch**
DCID-1
49.5 51.1
DCID-2
51.2 53.4
DCID-3
50.2 54.6 

**新预训练代码训练得到的模型，后面的数字代表epoch**
DCID-fast-new-1
52.0 50.8
DCID-fast-new-2
54.0 55.2
DCID-fast-new-3
49.4 53.7

如果你有需要，我可以提供训练得到的模型以及训练过程中的loss有关log给你。

**Comment by wy-coliney** [comment]
感谢你的工作和分享。有一点好奇的是你这里的DCID是指只用了DCID吗，还是说就是对应文中的完整的unicode？

请问我如何联系你获得新的模型？
![dcid](https://github.com/user-attachments/assets/80d8a61d-8f65-4ae9-be3a-adeff1097531)

**Comment by haihuangcode** [comment]
> 感谢你的工作和分享。有一点好奇的是你这里的DCID是指只用了DCID吗，还是说就是对应文中的完整的unicode？
> 
> 请问我如何联系你获得新的模型？ ![dcid](https://private-user-images.githubusercontent.com/30455008/392916618-80d8a61d-8f65-4ae9-be3a-adeff1097531.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0MTg0OTcsIm5iZiI6MTczMzQxODE5NywicGF0aCI6Ii8zMDQ1NTAwOC8zOTI5MTY2MTgtODBkOGE2MWQtOGY2NS00YWU5LWJlM2EtYWRlZmYxMDk3NTMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDE3MDMxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2YzU5MTU5ZjRmYzM3YjA3M2M0YmNjNzdjZDFmNWMzMGQzNDE5YTczYjdiOWExZTQ2NmViNDY4Nzc5NGE5NGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HeCiowyvs23lpuN-iwYff_ROENlwdocp5CJ5cIQO_Pw)

只用了DCID，新模型我白天整理下通过百度网盘发你。

**Comment by wy-coliney** [comment]
感谢分享。能顺带一下loss的log file和训练设置的参数吗？

**Comment by haihuangcode** [comment]
> 感谢分享。能顺带一下loss的log file和训练设置的参数吗？

log file我会加上的，训练设置的参数和现在github里的一样

**Comment by wy-coliney** [comment]
新的DCID就是直接使用更新的pre-train.py吗？还是说需要对main调整，因为我看到新的main中直接保存的就是DCID，所以我猜应该是直接使用就行？

**Comment by haihuangcode** [comment]
> 新的DCID就是直接使用更新的pre-train.py吗？还是说需要对main调整，因为我看到新的main中直接保存的就是DCID，所以我猜应该是直接使用就行？

直接用就行

**Comment by wy-coliney** [comment]
好的

**Comment by haihuangcode** [comment]
> 好的

通过网盘分享的文件：DCID-supplement
链接: https://pan.baidu.com/s/19S4xYsr8_iPTkpAnbKxigg?pwd=pgd9 提取码: pgd9

**Comment by wy-coliney** [comment]
谢谢分享

**Comment by haihuangcode** [comment]
非常不好意思，我今天才发现提供的vggsound-avel40k.csv是残缺的，里面只有20k条数据，应该是当时从服务器往本地传时出错了，今天才发现。经过测试，这应该就是你们训练出的模型和我训练出的有明显区别的原因，具体请参考更新的readme里的‘Model Checkpoints And Date Feature’新提供的'patch'文件。 @xueluofengfei @yyyxcleo
